src/plugins/intel_npu/src/compiler/CMakeLists.txt

```diff
@@ -0,0 +1,44 @@
+# Copyright (C) 2018-2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set(TARGET_NAME "openvino_npu_driver_compiler_adapter")
+
+file(GLOB_RECURSE SOURCES *.cpp *.hpp *.h)
+source_group(TREE ${CMAKE_CURRENT_SOURCE_DIR} FILES ${SOURCES})
+
+add_library(${TARGET_NAME} STATIC ${SOURCES})
+
+ov_build_target_faster(${TARGET_NAME}
+    PCH PRIVATE "src/precomp.hpp"
+)
+
+set_target_properties(${TARGET_NAME} PROPERTIES INTERPROCEDURAL_OPTIMIZATION_RELEASE ${ENABLE_LTO})
+ov_add_clang_format_target(${TARGET_NAME}_clang FOR_TARGETS ${TARGET_NAME})
+
+target_include_directories(${TARGET_NAME}
+    PUBLIC
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+)
+
+target_link_libraries(${TARGET_NAME}
+    PUBLIC
+        openvino::npu_al
+        ze_loader
+    PRIVATE
+        openvino_npu_zero_result_parser
+        openvino::runtime
+        openvino::runtime::dev
+)
+
+#
+# targets install
+#
+ov_install_static_lib(${TARGET_NAME} ${NPU_PLUGIN_COMPONENT})
+
+if(TARGET ze_loader)
+  install(TARGETS ze_loader EXPORT OpenVINOTargets
```

Comment:
because of it, we have https://github.com/openvinotoolkit/openvino/actions/runs/8482584927/job/23242437753?pr=23537

Reply:
Need this to pass static runtime, since we export openvino_intel_npu_plugin target, has to export the target it depends on. 

For the sample build issue, i think the reason is we export target with namespace, so ze_loader will be openvino::ze_loader when we try to build samples with OpenVIINOConfig.cmake. Add openvino::ze_loader in this file as alias of system ze_loader can solve issue. But if the ze_loader we found older than the one in our thirdparty, may still have link issue for unsolved symbols.  If version is compatible, the sample can build